packages/did/src/DidResolver/DidResolver.ts

```diff
@@ -33,129 +28,98 @@ import { getFullDidUri, parseDidUri } from '../Did.utils.js'
  */
 export async function resolve(did: DidUri): Promise<DidResolvedDetails | null> {
   const { type } = parseDidUri(did)
-  const fullDidUri = getFullDidUri(did)
-
-  switch (type) {
-    case 'full': {
-      const details = await Did.query(fullDidUri)
-      // If the details are found, return those details.
-      if (details) {
-        return {
-          details,
-          metadata: {
-            deactivated: false,
-          },
-        }
-      }
-      // If not, check whether the DID has been deleted or simply does not exist.
-      const isDeactivated = await queryDidDeletionStatus(did)
-      if (isDeactivated) {
-        return {
-          metadata: {
-            deactivated: true,
-          },
-        }
-      }
-      return null
+
+  const fullDid = await Did.query(getFullDidUri(did))
+  if (type === 'full' && fullDid) {
+    return {
+      details: fullDid,
+      metadata: {
+        deactivated: false,
+      },
+    }
+  }
+
+  // If the full DID has been deleted, return the info in the resolution metadata.
+  const isFullDidDeleted = await queryDidDeletionStatus(did)
```

Comment:
are we not only interested in this, if the `type` is `full`?

Reply:
ah, there was a duplication. I see. I think it would help to extend the comment, that even if a light did was migrated and then deleted, that we return the info in the metadata.

Reply:
Done in https://github.com/KILTprotocol/sdk-js/commit/a1257df987a4cf856c6e3ffe35673b75b2a2ff11